azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/async_internal_client.py

```diff
@@ -0,0 +1,95 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import logging
+import asyncio
+import functools
+from .internal_client import InternalClient
+
+logger = logging.getLogger(__name__)
+
+
+# all of our async wrappers can follow this same pattern.  We might be able to make this into a
+# helper function, but that will involve some tricky parameter handling because the signature of
+# callback changes depending on the function.
+
+# these are currently just functions in the module namespace.  We add them to the InternalClient
+# prototype below inside the monkeypatch() function.
+
+
+async def _connect_async(self):
+    """
+    This is just a test docstring to make sure we can see documentation for these functions
+    when we call help(InternalClient), and we can, so life is good and everyone is happy.
+    """
+    logger.info("async connecting to transport")
+
+    # we need the event loop so we can run our code in side it.
+    loop = asyncio.get_event_loop()
+
+    # the future is how we communicate the result from the callback task back into the "main" task
+    future = asyncio.Future()
```

Comment:
>future = asyncio.Future() [](start = 4, length = 25)

Is it not possible to use `loop.create_future()` ?

Reply:
that wasn't added until 3.7.  Do you know if there's any advantage to using that or disadvantage to doing it the way that I did?

Reply:
No particular advantage. 

---
In reply to: [238758002](https://github.com/Azure/azure-iot-sdk-python-preview/pull/28#discussion_r238758002) [](ancestors = 238758002)